src/datachain/query/session.py

```diff
@@ -64,25 +65,34 @@ def __init__(
 
         session_uuid = uuid4().hex[: self.SESSION_UUID_LEN]
         self.name = f"{name}_{session_uuid}"
-        self.job_id = os.getenv("DATACHAIN_JOB_ID") or str(uuid4())
         self.is_new_catalog = not catalog
         self.catalog = catalog or get_catalog(
             client_config=client_config, in_memory=in_memory
         )
+        self.versions: list[tuple[DatasetRecord, int]] = []
```

Comment:
Variable name `versions` is too broad, imo. Should we call it `datasets`, `dataset_versions` or may be `temp_datasets`? 👀